src/test/java/redis/clients/jedis/commands/jedis/ClusterCommandsTest.java

```diff
@@ -189,6 +201,34 @@ public void clusterSlots() {
     node1.clusterDelSlots(3000, 3001, 3002);
   }
 
+  @Test
+  public void clusterShards() {
+    assertEquals("OK", node1.clusterAddSlots(3100, 3101, 3102, 3105));
+
+    List<ClusterShardInfo> shards = node1.clusterShards();
+    assertNotNull(shards);
+    assertTrue(shards.size() > 0);
+
+    for (ClusterShardInfo shardInfo : shards) {
+      assertNotNull(shardInfo);
+
+      assertTrue(shardInfo.getSlots().size() > 1);
+      for (List<Long> slotRange : shardInfo.getSlots()) {
+        assertEquals(2, slotRange.size());
+      }
+
+      for (ClusterShardNodeInfo nodeInfo : shardInfo.getNodes()) {
+        assertNotNull(nodeInfo.getId());
+        assertNotNull(nodeInfo.getEndpoint());
+        assertNotNull(nodeInfo.getIp());
+        assertNotNull(nodeInfo.getPort());
+        assertNotNull(nodeInfo.getRole());
+        assertNotNull(nodeInfo.getHealth());
```

Comment:
Assert remaining fields.

Reply:
Assert that ltsPort and hostname is null and replicationOffset == 0, right?

Reply:
any sensible assert will do for now.
so, okay.

Reply:
Done